### ISSUE TITLE
🐛 BUG: Issues around handling of `<></>` Fragments

### ISSUE THREAD

**Issue by spaceemotion** [withastro/compiler-rs#46]
### What version of `@astrojs/compiler-rs` are you using?

0.2.2

### What package manager are you using?

pnpm

### What operating system are you using?

Linux

### Describe the Bug

_Originally filed as: https://github.com/withastro/astro/issues/17065_

It seems that the `@astrojs/compiler-rs` cannot handle the following syntax:

```jsx
{items.map((item) => <>
  <div>
    {item.label}
  </div>
</>)}
```

It somehow reports the line:col completely wrong for that kind of pattern, so it's not easy to spot what's wrong. Like how I am getting the following error for the same issue, with a completely wrong error message and source location:

```jsx
{item.cover && <>
  <Image
    src={item.cover}
    alt={``}
    aria-hidden="true"
    width={400}
    height={200}
    loading="lazy"
    decoding="async"
    fetchpriority="low"
    class="w-full h-32 sm:h-42 object-cover flex-none rounded-lg mb-1"
    style={{
      ...await getBlurryThumbnail(item.cover),
    }}
  />
</>}
```
showed up as:
```log
src/components/elements/RelatedContent.astro (21:39): Expected ',', got 'ident' (Note that you need plugins to import files that are not JavaScript)
file: /home/spaceemotion/code/nc-website/novelcrafter/src/components/elements/RelatedContent.astro:21:39

19:   },
20:
21:   courses: {
                ^
22:     label: 'Course',
23:   },
```

The following fix seems to work, but it seems weird that I cannot just have a fragment like that?
```jsx
{items.map((item) => (
  <div>
    {item.label}
  </div>
)}
```
```jsx
{item.cover && (
  <Image
    // ...
  />
)}
```

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-djctpt5w?file=src%2Fpages%2Findex.astro
